### DISCUSSION TITLE
SIGTSTP (Control-Z) ignored during Batect running

### DISCUSSION THREAD

**Discussion by binkley** [batect/batect#1139] [Q&A]
I find that Control-Z (SIGTSTP "terminal stop", ie, suspend foreground process) is sometimes ignored (likely the process is catching and ignoring the signal) while running Batect.

**Any suggestions how to mitigate this?**

In principle I could open another terminal, and send SIGSTOP "suspend, damn you!", but that's a nuisance.

I had to go reread up on the job control signals to understand this (https://en.wikipedia.org/wiki/Job_control_(Unix)):
```
A job running in the foreground can be stopped by typing the suspend character (Ctrl-Z). This sends the "terminal stop" signal (SIGTSTP) to the process group. By default, SIGTSTP causes processes receiving it to stop, and control is returned to the shell. However, a process can register a signal handler for or ignore SIGTSTP. A process can also be paused with the "stop" signal (SIGSTOP), which cannot be caught or ignored.
```

**Comment by charleskorn** [comment]
Do you see the same behaviour when running a container with the `docker` CLI directly?
